router/router.go

```diff
@@ -242,21 +243,21 @@ func New(cfg *config.Configuration, rateConvertor *currencies.RateConverter) (r
 
 	exchanges = newExchangeMap(cfg)
 	cacheClient := pbc.NewClient(cacheHttpClient, &cfg.CacheURL, &cfg.ExtCacheURL, r.MetricsEngine)
-	theExchange := exchange.NewExchange(generalHttpClient, cacheClient, cfg, r.MetricsEngine, bidderInfos, gdprPerms, rateConvertor)
+	theExchange := exchange.NewExchange(generalHttpClient, cacheClient, cfg, r.MetricsEngine, bidderInfos, gdprPerms, rateConvertor, categoriesFetcher)
```

Comment:
Maybe accounts fetcher could also go in here, since it's not endpoint-specific either.

Reply:
Perhaps. Let's explore that in a follow-up PR. The endpoints are each performing the account lookup right now with some, such as AMP, doing non-trivial things. I started on the refactor for this PR but it got a bit complicated.